src/Compilers/Core/Portable/Compilation/SemanticModel.cs

```diff
@@ -876,5 +876,15 @@ protected internal virtual SyntaxNode GetTopmostNodeForDiagnosticAnalysis(ISymbo
         {
             return declaringSyntax;
         }
+
+        /// <summary>
+        /// Root of this semantic model
+        /// </summary>
+        internal SyntaxNode Root => RootCore;
+
+        /// <summary>
+        /// Root of this semantic model
+        /// </summary>
+        protected abstract SyntaxNode RootCore { get; }
```

Comment:
We should follow the existing style which seems to be `protected` rather than `internal`. (I saw a couple of `internal` Core members when I looked earlier and assumed `internal` was preferred but perhaps not.)